### PR TITLE
En 1131 user role detail

### PR DESCRIPTION
**Pull request by alessioscalici**

### CODE REVIEW

**Comment by mcarta-rt** [comment]
retest this please

**Comment by alessioscalici** [comment]
retest this please

**Comment by entando-service** [comment]
retest this please

**Comment by entando-service** [comment]
ok to test

**Comment by entando-service** [comment]
retest this please

File: src/state/roles/selectors.js [hunk 1]
```diff
@@ -1,10 +1,25 @@
...
+export const getUserRefs = state => state.roles.selected.userReferences;
 
 export const getRolesList = createSelector(
   [getRolesMap, getRolesIdList],
   (rolesMap, idListRoles) => idListRoles.map(id => (rolesMap[id])),
 );
+
+export const getSelectedRolePermissionsList = createSelector(
+  [getSelectedRolePermissions, getPermissionsList],
+  (rolePermissions, permissions) => permissions.filter(permission =>
```
Comment (nicpuddu) [inline src/state/roles/selectors.js]: there's no need to use square brackets for this

File: src/state/roles/selectors.js [hunk 2]
```diff
@@ -1,10 +1,25 @@
...
+    rolePermissions[permission.code]).map(permission => permission.descr),
+);
+
+export const getSelectedRole = createSelector(
+  [getSelectedRolePermissionsList, getRoles],
+  (permissions, roles) => ({ ...roles.selected, permissions }),
```
Comment (nicpuddu) [inline src/state/roles/selectors.js]: same applies here

File: src/ui/common/references/UserRefsTable.js [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+  userReferences: PropTypes.arrayOf(PropTypes.shape({
+    username: PropTypes.string,
+    fullName: PropTypes.string,
+    lastLogin: PropTypes.string,
+    status: PropTypes.string,
+  })),
+};
+
+UserRefsTable.defaultProps = {
+  onWillMount: null,
```
Comment (nicpuddu) [inline src/ui/common/references/UserRefsTable.js]: in the onWillMount method of the component it does not check whether or not onWillMount is defined before calling it as a function, although the default is null.
If the parameter is being passed by the container anyway it should just be required and have no default, or if the check does not happen it should default to an empty function

**Pull request merged**